llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -2341,6 +2373,28 @@ class VPReductionRecipe : public VPRecipeWithIRFlags {
     setUnderlyingValue(I);
   }
 
+  /// For VPExtendedReductionRecipe.
+  /// Note that IsNonNeg flag and the debug location are from the extend.
+  VPReductionRecipe(const unsigned char SC, const RecurKind RdxKind,
+                    ArrayRef<VPValue *> Operands, VPValue *CondOp,
+                    bool IsOrdered, NonNegFlagsTy NonNeg, DebugLoc DL)
+      : VPRecipeWithIRFlags(SC, Operands, NonNeg, DL), RdxKind(RdxKind),
+        IsOrdered(IsOrdered), IsConditional(CondOp) {
+    if (CondOp)
+      addOperand(CondOp);
+  }
+
+  /// For VPMulAccumulateReductionRecipe.
+  /// Note that the NUW/NSW and DL are from the Mul.
```

Comment:
```suggestion
  /// Note that the NUW/NSW flags and the debug location are from the Mul.
```

(for consistency with comment above)